packages/ember-metal/lib/utils.js

```diff
@@ -105,6 +105,14 @@ function intern(str) {
   return str;
 }
 
+export function symbol(debugName) {
+  // TODO: Investigate using platform symbols, but we do not
+  // want to require non-enumerability for this API, which
+  // would introduce a large cost.
+
+  return intern(debugName + ' [id=' + GUID_KEY + Math.floor(Math.random() * new Date()) + ']');
```

Comment:
I think you're actually increasing the odds of collision significantly, not reducing them.

"Not cryptographically strong" just means that it's possible to make better-than-expected guesses about future values based on past values. It doesn't mean the distribution is skewed. `Math.random()` is supposed to give a reasonably uniform distribution over 64 bits.

When you multiply by date and cast down to integer, you throw away roughly 24 bits worth of entropy, because the date is only going to be roughly `2^40`. So you end up with only the integers integers from 0 to 2^40.

If instead you did roughly:

    var MAX_DOUBLE = Math.pow(2, 64);
    return Math.floor(Math.random() * MAX_DOUBLE);

you will get integers from 0 to 2^64.